docs/articles/documentation/test-api/test-code-structure.md

```diff
@@ -41,166 +39,282 @@ This function returns the `fixture` object that allows you to configure the fixt
 
 > [Tests](#tests) that constitute a fixture go after this declaration.
 
-### Specifying the Start Webpage
+## Tests
+
+To introduce a test, call the `test` function and pass test code inside.
+
+```text
+test( testName, fn(t) )
+```
+
+Parameter  | Type     | Description
+---------- | -------- | --------------------------------------------------------------------
+`testName` | String   | The test name.
+`fn`       | Function | An asynchronous function that contains test code.
+`t`        | Object   | The [test controller](#test-controller) used to access test run API.
+
+```js
+fixture `MyFixture`;
+
+test('Test1', async t => {
+    /* Test 1 Code */
+});
+
+test('Test2', async t => {
+    /* Test 2 Code */
+});
+```
+
+You are free to structure test code in any manner, and you can
+reference any modules or libraries you need.
+
+TestCafe tests are executed server side. To manipulate the tested webpage, use [test actions](actions/index.md).
+To determine the state of page elements or obtain any other data from the client side, use [selectors](selecting-page-elements/selectors.md) and
+[client functions](obtaining-data-from-the-client.md).
+
+To check if the page state matches the expected one, use [assertions](assertions/index.md).
+
+### Test Controller
+
+A *test controller* object `t` exposes methods of test API. That is why it is
+passed to each function that is expected to contain server-side test code (like [test](#tests),
+[beforeEach](#initialization-and-clean-up) or [afterEach](#initialization-and-clean-up)).
+
+Use the test controller to call [test actions](actions/index.md), handle [browser dialogs](handling-native-dialogs.md),
+use the [wait function](pausing-the-test.md) or [invoke selectors](selecting-page-elements/index.md).
+
+```js
+fixture `My fixture`
+    .page `http://www.example.com/`;
+
+test('My Test', async t => {
+    await t
+        .setNativeDialogHandler(() => true)
+        .click('#populate')
+        .click('#submit-button');
 
-You can optionally use the fixture declaration to specify a webpage at which all tests in a fixture start.
-To do this, use the `page` function.
+    const location = await t.eval(() => window.location);
+
+    await t.expect(location.pathname).eql('/testcafe/example/thank-you.html');
+});
+```
+
+Another job of the test controller is providing access to the internal context required for test API to operate.
+This is why [selectors](selecting-page-elements/selectors.md) and
+[client functions](obtaining-data-from-the-client.md) need the test controller object when they are
+called from Node.js callbacks.
+
+## Specifying the Start Webpage
+
+You can specify a webpage at which all tests in a fixture start.
+To do this, use the `fixture.page` function.
 
 ```text
-page( url )
+fixture().page( url )
+fixture.page( url )()
```

Comment:
I don't like variants without fixtureName and testBody, because now it looks too complicated